lib/cli-engine.js

```diff
@@ -694,6 +690,8 @@ CLIEngine.prototype = {
                     descriptor.meta.hashOfConfig = hashOfConfig;
                     descriptor.meta.results = res;
                 }
+            } else {
```

Comment:
Question: Is there a reason you put this here instead of the first `if` block (https://github.com/eslint/eslint/pull/6540/files#diff-3daec82d5218d64d8f2fed4d21f8e6ccR636) ?